packages/flutter/test/widgets/media_query_test.dart

```diff
@@ -262,6 +262,57 @@ void main() {
     },
   );
 
+  testWidgets('MediaQueryData.fromView picks up new view data when rebuilding', (
+    WidgetTester tester,
+  ) async {
+    addTearDown(() {
+      tester.platformDispatcher.clearAllTestValues();
+    });
+
+    final List<MediaQueryData> datas1 = <MediaQueryData>[];
+
+    await tester.pumpWidget(
+      MediaQuery.fromView(
+        view: tester.view,
+        child: Builder(
+          builder: (BuildContext context) {
+            datas1.add(MediaQuery.of(context));
+            return const Placeholder();
+          },
+        ),
+      ),
+    );
+
+    expect(datas1, hasLength(1));
+    expect(datas1.first.supportsShowingSystemContextMenu, isFalse);
+
+    tester.platformDispatcher.supportsShowingSystemContextMenu = true;
+    tester.binding.handleSupportsShowingSystemContextMenuChanged();
+    addTearDown(() {
```

Comment:
Can these be added to the `addTearDown` at the top of this test?

Reply:
Yes good call.